float32[3] control_output